### PR TITLE
TCM-368 | Polish plans screen

### PR DESCRIPTION
**Pull request by vespertineSiren**
### Summary

- Follow [figma](https://www.figma.com/file/LXa0oBt9aSNOk0YMAeSM8Q/TRX-Consumer?node-id=1703%3A2) for `Plans` from ticket. 
- Make sure the data is connected. 

### Issue

[TCM-368](https://hyfnla.atlassian.net/browse/TCM-368)

### Video/Screenshot

Prod Login: `ashley+prod3@commaeight.com`

https://user-images.githubusercontent.com/27742105/120122997-5bbc7d00-c161-11eb-8735-31e929e2c817.mov

| No Plan | Plan | 
|----|----|
|Screenshot_1622427563|Screenshot_1622427398|

### CODE REVIEW

File: app/src/main/java/com/trx/consumer/screens/plans/PlansFragment.kt [hunk 1]
```diff
@@ -62,6 +62,7 @@ class PlansFragment : BaseFragment(R.layout.fragment_plans) {
         LogManager.log("handleLoadCanCancel")
         viewBinding.apply {
             btnCancel.isHidden = !value
+            btnCancel.action { viewModel.doTapUnsubscribe() }
```
Comment (VivekV95) [inline app/src/main/java/com/trx/consumer/screens/plans/PlansFragment.kt:65]: Should we show the cancel alert here?

File: app/src/main/java/com/trx/consumer/models/responses/PlansResponseModel.kt [hunk 2]
```diff
@@ -11,15 +11,19 @@ class PlansResponseModel(private var plan: PlansModel = PlansModel()) {
...
                     list.removeAt(index).let { planModel ->
+                        list.clear()
                         planModel.primaryState = PlansViewState.CURRENT
                         list.add(0, planModel)
                     }
                 }
             }
             list
-        } ?: plan.plans
+        } ?: listOf()
```
Comment (VivekV95) [inline app/src/main/java/com/trx/consumer/models/responses/PlansResponseModel.kt:26]: Screen Shot 2021-05-30 at 8 53 53 PM

Looks like we should be doing `userModel?.planText?.let { }` here in `PlansViewModel`, which means that `planText` can be passed in as non-null and hence no need to the null handling

**Review by VivekV95 — CHANGES_REQUESTED** [review]
Couple thoughts

File: app/src/main/java/com/trx/consumer/models/responses/PlansResponseModel.kt (hunk 2 above)
Comment (vespertineSiren) [reply app/src/main/java/com/trx/consumer/models/responses/PlansResponseModel.kt]: Here we are: https://github.com/hyfn/trx-consumer-android/pull/89/commits/38a446e8b75742298e909262db9a924e26fb7499

File: app/src/main/java/com/trx/consumer/screens/plans/PlansFragment.kt (hunk 1 above)
Comment (vespertineSiren) [reply app/src/main/java/com/trx/consumer/screens/plans/PlansFragment.kt:66]: I have no preference either way

Comment (VivekV95) [reply app/src/main/java/com/trx/consumer/screens/plans/PlansFragment.kt:66]: Think that'd be safer and more consistent with our usual transaction stuff, user could easily hit the button accidentally. Not sure what iOS is currently doing

Comment (vespertineSiren) [reply app/src/main/java/com/trx/consumer/screens/plans/PlansFragment.kt:66]: Theyre doing this weird thing that I was sure on how to implement. 
`self.viewMain.btnCancelPlan.action{ self.handleLoadCancelPlan(CacheManager.shared.user?.plan) }`

Comment (VivekV95) [reply app/src/main/java/com/trx/consumer/screens/plans/PlansFragment.kt:66]: Yeah looks like that ends up showing the cancel alert, we'll just have to create a `ViewModel` function like `doLoadCancelPlan` or something

Comment (vespertineSiren) [reply app/src/main/java/com/trx/consumer/screens/plans/PlansFragment.kt]: Here we go: https://github.com/hyfn/trx-consumer-android/pull/89/commits/335bdeac784a8e390ed7094adcd0de17c63bf25f

**Review by onodude — APPROVED** [review]
thank you!

**Pull request merged**